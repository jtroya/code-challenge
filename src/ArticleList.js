import React from 'react';
import PropTypes from 'prop-types';

function ArticleList({ items }) {
  return (
    <div className="container">
      { items.map(item => (
        <div key={item.id} className="box">
          <h3 className="box-title">{item.author}</h3>
          <p className="box-excerpt">{item.excerpt}</p>
        </div>))
      }
    </div>
  );
}

ArticleList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape(
    {
      id: PropTypes.string,
      author: PropTypes.string,
      excerpt: PropTypes.string,
    })),
};

ArticleList.defaultProps = {
  items: [],
};

export default ArticleList;
