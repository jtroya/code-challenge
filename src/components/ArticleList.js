import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class ArticleList extends Component {
  componentDidMount() {
    this.props.fetchArticles();
  }

  render() {
    const { articles, loading } = this.props;
    return (
      <div className="container">
        { loading &&
          <div className="loading">
            <h3 className="box-title">loading</h3>
          </div>
        }
        { !loading && articles.map(item => (
          <div key={item.id} className="box">
            <Link to={`/article/${item.id}`} className="link-article">
              <h3 className="box-title">{item.author}</h3>
            </Link>
            <p className="box-excerpt">{item.excerpt}</p>
          </div>)) }
      </div>
    );
  }
}

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape(
    {
      id: PropTypes.string,
      author: PropTypes.string,
      excerpt: PropTypes.string,
    })),
  fetchArticles: PropTypes.func,
  loading: PropTypes.bool,
};

export default ArticleList;
