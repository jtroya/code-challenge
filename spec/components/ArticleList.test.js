import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';

import ArticleList from '../../src/components/ArticleList';

describe('ArticleList', () => {
  let wrapper = null;
  const fetchArticles = jest.fn();
  const fakeArticles = [{ id: 'ABC123', title: 'my title' }];

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <ArticleList
          articles={fakeArticles}
          loading={false}
          fetchArticles={fetchArticles}
        />
      </MemoryRouter>,
    );
  });

  it('should render the component', () => {
    expect(wrapper.find('ArticleList')).toBeTruthy();
  });

  it('should render Loading component', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper.find('.loading')).toBeTruthy();
  });

  it('should matches the snapshot', () => {
    const tree = shallow(
      <ArticleList
        articles={fakeArticles}
        loading={false}
        fetchArticles={fetchArticles}
      />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
