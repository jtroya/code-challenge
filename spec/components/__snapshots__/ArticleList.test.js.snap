exports[`ArticleList should matches the snapshot 1`] = `
<div
  className="container">
  <div
    className="box"
    key="ABC123">
    <Link
      className="link-article"
      replace={false}
      to="/article/ABC123">
      <h3
        className="box-title" />
    </Link>
    <p
      className="box-excerpt" />
  </div>
</div>
`;
